src/java/org/tonyzampogna/xss/sanitizer/util/XssSanitizerUtil.java

```diff
@@ -57,7 +57,7 @@
 			XSS_INPUT_PATTERNS.add(Pattern.compile("onfocus(.*?)=", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL));
 
 			// Avoid any form injection with <...form ...> ... </form ...> tag
-			XSS_INPUT_PATTERNS.add(Pattern.compile("<(.*?)form(.*?)>(.*?)</(.*?)form(.*?)>", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL));
+			XSS_INPUT_PATTERNS.add(Pattern.compile("<([^<>]*)form([^<>]*)>(.*?)</([^<>]*)form([^<>]*)>", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL));
```

Comment:
Can the regex expression stay "lazy". For example,

    Pattern.compile("<([^<>]*?)form([^<>]*?)>....

Also, you may want to submit your change to the Grails 3 plugin.

http://plugins.grails.org/plugin/rpalcolea/xss-sanitizer

Reply:
Having revisited this, I agree that the groups should use a lazy quantifier. The change I originally made was a direct copy & paste from the Grails 3 plugin you reference. It does feel like the Grails 3 plugin should pick up the end result of this PR for consistency.

One observation I have while experimenting with this is that the forward slash in the latter half of the pattern must be being ignored as invalid since it is not escaped. If it were not being ignored then I don't think our sample XML would have matched. It turns out you need two elements of our data to trigger the match with the old regex and no line break between them. For example:

```
<tag expression="payload.Data.Initiation.RemittanceInformation"  gen="jag"/><tag expression="payload.Data.Initiation.RemittanceInformation.Unstructured"  gen="jag"/>
```